### DISCUSSION TITLE
Terminal API documentation

### DISCUSSION THREAD

**Discussion by pjmagee** [dagger/dagger#12272] [Ideas]
### What is the issue?

After a discussion on 'wait for user input' here: https://discord.com/channels/707636530424053791/1277921913058099210

I mentioned some suggestions around scripting or securing the dagger command OUTSIDE of dagger execution. However user input reminded me very much of the `terminal()` feature. Reading through the dagger documentation, the `terminal()` is documented to be very much only as a DX for debugging, because in many cases, the `terminal()` would not be useful or execute properly when running on a remote CI environment, as it cannot take input or connect you to the terminal. 

However, this does not mean that `terminal()` could not be better utilised for local usage, one example we came up with was reading user input and deciding if the dagger pipeline should continue or not. This isn't only debugging, this is now using `terminal()` as a local development feature.

Can we better document some other use-cases for the terminal documentation, and maybe describe it's weaknesses or places to avoid using it and other suggestions of how the terminal could be used within a pipeline? Possibly using this 'wait for user input' as an example?

**Comment by jedevc** [comment]
(deleted spam :broom:)
